apps/meteor/ee/server/lib/ldap/Manager.ts

```diff
@@ -54,16 +55,28 @@ export class LDAPEEManager extends LDAPManager {
 			const membersOfGroupFilter = await ldap.searchMembersOfGroupFilter();
 
 			await converter.convertData({
-				beforeImportFn: (async ({ options }: IImportRecord): Promise<boolean> => {
-					if (!ldap.options.groupFilterEnabled || !ldap.options.groupFilterGroupMemberFormat) {
+				beforeImportFn: (async ({ options, data }: IImportRecord): Promise<boolean> => {
+					if (ldap.options.groupFilterEnabled && ldap.options.groupFilterGroupMemberFormat) {
+						const memberFormat = ldap.options.groupFilterGroupMemberFormat
+							?.replace(/#{username}/g, options?.username || '#{username}')
+							.replace(/#{userdn}/g, options?.dn || '#{userdn}');
+
+						if (!membersOfGroupFilter.includes(memberFormat)) {
+							return false;
+						}
+					}
+
+					const userExists = await converter.findExistingUser(data as IImportUser);
+					if (userExists) {
 						return true;
 					}
 
-					const memberFormat = ldap.options.groupFilterGroupMemberFormat
-						?.replace(/#{username}/g, options?.username || '#{username}')
-						.replace(/#{userdn}/g, options?.dn || '#{userdn}');
+					if (await License.shouldPreventAction('activeUsers', 1)) {
```

Comment:
Adding a change request here just to block the PR from being merged;

We should not check the license before every operation.